### PR TITLE
Ignore multiple events...

### PR DESCRIPTION
**Pull request by crocri**
Hi!
Please have a look at my implementation of suppress multiple events. It's configurable for the user and not fixed to 25 seconds.
Also the time is restarted, if event comes in during the timeout.
Here an example for 15 seconds:
first event arrives
second after 10 seconds - ignored
third after 20 seconds - also ignored, because timer restarts after second one.
fourth after 35 seconds - is executed
and so on...

Your decision, which one you want to have :-)

### CODE REVIEW

**Comment by theimo1221** [comment]
Hi @crocri 

Whilst having it configurable is definitely good, your solution with object assign is a big anti pattern in Typescript.

Additionally you have a potential bug due to clearing of the timeout without resetting it in nlrmaljexecution, as Node can recycle those id's after the timer did fire.

If it is okay for you I'll modify your PR to mitigate the bug and change it to match best practices.

Thanks and best regards

Thiemo

**Comment by theimo1221** [comment]
Hello @crocri 

Did some updates and prepared an option to follow up on your idea 

> also ignored, because timer restarts after second one.

This option is not yet included in the config page and translations as I'd like your feedback on this approach first.
In my opinion this has to be configurable, as Ring is no 24/7 recording, meaning some users definetly want to catch every event, while still they want to debounce the doorbell by e.g. 20s. Meaning they don't want several events within the 20s, but I'm not sure if they'd be happy, if a double press (0s --> 10s) would result in another press after 29s being ignored.

Additionally if somebody keeps dancing in front of a camera for like 5 Minutes straight, it might be beneficial to have e.g. 10 videos (one every 30s) instead of really just one, but sure this depends on the user and his needs/wants.

Best regards

Thiemo

File: src/lib/ownRingCamera.ts [hunk 1]
```diff
@@ -1140,81 +1140,101 @@ export class OwnRingCamera extends OwnRingDevice {
...
-      this._adapter.upsertState(
-        `${this.eventsChannelId}.doorbell`,
-        COMMON_EVENTS_DOORBELL,
-        false,
-      );
-    }, 5000);
-    setTimeout(() => this._doorbellEventActive = false, 25000);
-    this.conditionalRecording(EventState.ReactingOnDoorbell, value.ding.image_uuid);
-  }
+  private onDing = Object.assign(
```
Comment (theimo1221) [inline src/lib/ownRingCamera.ts:1157]: @crocri why did you revert to Object.assign and didn't proceed with my variant from commit https://github.com/iobroker-community-adapters/ioBroker.ring/pull/604/commits/2211704ece3c315914d6e24c3593262e2b9d9197 ?

I still have mayor concerns about type safety and best practices with this approach

**Review by theimo1221 — CHANGES_REQUESTED** [review]

File: src/lib/ownRingCamera.ts (hunk 1 above)
Comment (crocri) [reply src/lib/ownRingCamera.ts:1157]: Hi!
It's more elegant? If the types defined correct, I see no problems. I also think about date function to get actual time and compare with old one. But to repeat this if event is in time range, it's much more complicated to implement. You also need a "private" variable to set old time...
Further on I see no problems with Timeout id, It's reserverd, then cleared and afterwards a new one is reserved.

Comment (theimo1221) [reply src/lib/ownRingCamera.ts]: While I agree on a combined handler reducing duplicate code, thus reducing maintenance, I hardly disagree with object.assign being more elegant in a typescript enviroment. You could create a new class, you could extract a blocking-mechanism, but to override a property which should only be a function, is in my opinion a pitfall for future development.

**Pull request merged**